sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -788,6 +790,28 @@ static Pair<DataSource, Filtration> getFiltration(
   {
     if (!canUseIntervalFiltering(dataSource)) {
       return Pair.of(dataSource, toFiltration(filter, virtualColumnRegistry.getFullRowSignature(), false));
+    } else if (dataSource instanceof UnnestDataSource) {
```

Comment:
nit: It won't make a difference with the logic as is, but let's put this branch at the top since it is the most specific of the branches.